### ISSUE TITLE
Unable to deploy VM using ISO and custom disk offering

### ISSUE THREAD

**Issue by sureshanaparti** [apache/cloudstack#6440]
Deploy VM using ISO,and custom disk offering fails with error "VM Creation failed. Volume size: -1GB is out of allowed range. Max: 1024 Min:1".

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
VM
~~~

##### CLOUDSTACK VERSION

~~~
4.17.0.0 RC4
~~~

##### CONFIGURATION

Hypervisor: KVM / VMware, Advance Networking

##### OS / ENVIRONMENT

MS on CentOS 7

##### SUMMARY

Unable to deploy VM using ISO and custom disk offering, fails with error "VM Creation failed. Volume size: -1GB is out of allowed range. Max: 1024 Min:1". 

<img alt="DeployVM_ISO_KVM_4 17 0 RC4" src="https://user-images.githubusercontent.com/12028987/172117605-11c66dee-5de3-42dd-aa67-a99fffd5dfeb.png" width="70%" height="70%">

<img alt="DeployVM_ISO_VMware_4 17 0 RC4" src="https://user-images.githubusercontent.com/12028987/172117625-01cdd0a5-f943-4927-bfa6-6a74f131dde3.png" width="70%" height="70%">

Able to deploy VM using ISO and custom disk in 4.16.1.

deployVirtualMachine API inputs from UI:

- 4.16.1

```
zoneid: a03c812a-4c47-4ec8-aa23-9538f89f23e5
boottype: BIOS
bootmode: LEGACY
templateid: 7eaa67ca-ecf7-49f7-8852-19ac500d4061
hypervisor: KVM
startvm: true
serviceofferingid: 488a4f2a-2260-4dbd-bd22-3e53e464ee4e
diskofferingid: 2ab2a27c-acf0-4bb2-9352-0bb0e1151d67
size: 10
affinitygroupids: 
iptonetworklist[0].networkid: 0bc5477c-b2b1-4399-bac2-4c8816a7c723
name: testisovm03
displayname: testisovm03
```

- 4.17.0 RC4

```
zoneid: 8a2cc70b-1afd-4bb2-bcd6-5b5e80272351
boottype: BIOS
bootmode: LEGACY
templateid: 59d0d9c5-8f9e-4876-a138-eb82d232efc7
hypervisor: KVM
startvm: true
serviceofferingid: b71294ec-89cd-4bff-be1e-b571ea05e76d
diskofferingid: e9952a98-2a71-491c-bfeb-4e06d32bce74
size: 10
affinitygroupids: 
iptonetworklist[0].networkid: e59106c4-1cc6-450c-82f7-d429665eb68c
keypairs: 
name: testisovm03-417
displayname: testisovm03-417
```

Also, noticed that hypervisor is not selected by default in 4.17.0, whereas 4.16.1 selects hypervisor (is this intentional?)

<img alt="DeployVM_ISO_HypervisorSelection_4 17 0 RC4" src="https://user-images.githubusercontent.com/12028987/172117647-db94de17-5d77-4946-a1d4-24bef41daed1.png" width="70%" height="70%">

##### STEPS TO REPRODUCE

- Register ISO
- Create custom disk offering
- Deploy VM using ISO and custom disk offering (specify any size, eg. 10 GB)

##### EXPECTED RESULTS

- Deploy VM using ISO and using custom offering fails.

##### ACTUAL RESULTS

- Deploy VM using ISO and using custom offering should succeed.

**Comment by nvazquez** [comment]
Removing the blocker tag, since there are 2 workarounds:
- Pass the 'rootdisksize' parameter to the deployVM API
- Use a fixed offering with the expected size

**Comment by nvazquez** [comment]
Fixed on #6441
